brewchat/src/main/java/com/example/brewchat/ChatService.java

```diff
@@ -345,22 +398,37 @@ public void processUserStopTyping(Object o) {
 
     @Override
     public void processMessage(QBChat qbChat, QBChatMessage qbChatMessage) {
-        User u = null;
+        Log.d(TAG, "processMessage: " + qbChat + " " + qbChatMessage);
+        // TODO THIS IS A MESS! CLEAN IT BEFORE PUSHING!
+        QBUser sender = null;
         try {
-            QBUser sender = getUserById(qbChatMessage.getSenderId());
-            u = new User();
-            u.setId(sender.getId());
-            u.setLastRequestAt(u.getLastRequestAt());
-            u.setLogin(sender.getLogin());
-            u.setEmail(sender.getEmail());
-            u.setName(sender.getFullName());
+            sender = getUserById(qbChatMessage.getSenderId());
         } catch (QBResponseException e) {
             e.printStackTrace();
             return;
         }
-        // Bracket hell ->
-        EventBus.getDefault().post(new MessageReceivedEvent(new ChatMessage(u, qbChatMessage.getBody())));
-        Log.d(TAG, "processMessage: " + qbChat + " " + qbChatMessage);
+        User u = new User();
+        u.setId(sender.getId());
+        u.setLastRequestAt(u.getLastRequestAt());
+        u.setLogin(sender.getLogin());
+        u.setEmail(sender.getEmail());
+        u.setName(sender.getFullName());
+        if (qbChat instanceof QBPrivateChat) {
+            try {
+                ((QBPrivateChat) qbChat).readMessage(qbChatMessage.getId());
+            } catch (XMPPException | SmackException.NotConnectedException e) {
+                // If there's a problem, it's not a big deal. we can still continue on.
```

Comment:
The readMessage method marks the method as read. It's not a big deal if there's an error making that mark. Just means we'll receive the message again when we reconnect and can retry then.